### DISCUSSION TITLE
How to use with Remix Run

### DISCUSSION THREAD

**Discussion by mtzfactory** [tamagui/tamagui#143] [General]
I would like use this library within a [Remix Run](https://remix.run/) web, but can figure it how.

Any help would be appreciated.

**Comment by frontsideair** [comment]
I think it's not possible to use Tamagui with Remix, at least not without significant effort. Tamagui uses webpack while Remix uses esbuild. Theoretically, you may run a separate process to run Tamagui compiler, like how Remix uses Tailwind in official examples. The ideal scenario would be to wait for Remix or Tamagui to work on integration.

**Comment by natew** [comment]
If someone starts a PR, I'm happy to help that along. The webpack loader and babel plugin both share a base `@tamagui/static` package that abstracts a lot of the logic out.

**Comment by penx** [comment]
Perhaps something along the lines of the remix vanilla-extract example will work?

> A separate [tsup](https://github.com/egoist/tsup) process is needed to generate CSS
https://github.com/remix-run/examples/tree/main/vanilla-extract

**Comment by penx** [comment]
I am pretty sure the Remix team want to get CSS Modules and better vanilla extract support in, but they're taking their time to get it right.

It would be worth keeping an eye on these discussions, as any progress on Vanilla Extract and CSS Modules integration in Remix would likely open up options for tamagui too

https://github.com/remix-run/remix/discussions/2214
https://github.com/remix-run/remix/discussions/2567

**Comment by alexluong** [comment]
Hi, I'm curious if there's any development on this effort since the issues mentioned by @penx above have been resolved?

**Comment by ryan0x44** [comment]
It looks like Remix now fully supports Vanilla Extract and CSS-in-JS in general, so it should be possible to support Tamagui? https://remix.run/docs/en/1.18.0/guides/styling#css-in-js-libraries

**Comment by ryan0x44** [comment]
currently have a somewhat working version here: https://github.com/ryan0x44/remix-tamagui-new/commits/main

more details about this on discord discussion here: https://discord.com/channels/909986013848412191/1078243378627616798/1131207088866070628

**Reply by alexluong** [reply]
Amazing! Responded to the thread on Discord, would be happy to help out further if there's anything I can do!
